### ISSUE TITLE
kernel keeps dying

### ISSUE THREAD

**Issue by inktrap** [gibiansky/IHaskell#655]
Maybe this is a problem caused by me, but ihaskell on debian testing does not work for me.

I installed it via cabal (see below) and registered the kernel. I get the following message that cryptonite is not installed (``ihaskell: ihaskell: unable to load package `cryptonite-0.8'``) but it is.

# steps

Here are the detailed steps that I did:

 - prerequisites: a working haskell requirement: ``sudo apt-get install haskell-platform ghc``
 - remove old config: ``rm -rf .config/ipython``
 - install ipython3: ``sudo pip3 install jupyter ipython jinja2 zmq tornado``
 - start the notebook to see if it works: ``jupyter notebook --ip=127.0.0.1``
 - install zeromq-3: ``sudo apt-get install libzmq3-dev``
 - if it works, install ihaskell via cabal: ``cabal install ihaskell --reorder-goals``
 - ``sudo ln -s $(which ipython3) /usr/local/bin/ipython``
 - clear any alias that you might have set for ipython or jupyter
 - add the new kernel to ipython: ``ihaskell install``
 - start the notebook to see if it works: ``jupyter notebook --ip=127.0.0.1``
 - status: haskell kernel keeps dying

That didn't work, so I unregistered all the packages and tried it via the install script from the repo (``ubuntu-install.sh``). However the result is the same and the kernel keeps dying.

 - removed all installations via pip and installed via git and scripts, same problem

# Errorlogs

## console

```
vh@box ~ [n] % jupyter console --debug --kernel haskell
[ZMQTerminalIPythonApp] Searching ['/home/vh', '/home/vh/.jupyter', '/usr/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[ZMQTerminalIPythonApp] Looking for jupyter_config in /etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_config in /usr/local/etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_config in /usr/etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_config in /home/vh/.jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_config in /home/vh
[ZMQTerminalIPythonApp] Looking for jupyter_console_config in /etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_console_config in /usr/local/etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_console_config in /usr/etc/jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_console_config in /home/vh/.jupyter
[ZMQTerminalIPythonApp] Looking for jupyter_console_config in /home/vh
[ZMQTerminalIPythonApp] Connection File not found: /run/user/1000/jupyter/kernel-24987.json
[ZMQTerminalIPythonApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[ZMQTerminalIPythonApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[ZMQTerminalIPythonApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-24987.json', '--debug', '--ghclib', '/usr/lib/ghc']
[ZMQTerminalIPythonApp] Connecting to: tcp://127.0.0.1:37707
[ZMQTerminalIPythonApp] connecting shell channel to tcp://127.0.0.1:56017
[ZMQTerminalIPythonApp] Connecting to: tcp://127.0.0.1:56017
[ZMQTerminalIPythonApp] connecting iopub channel to tcp://127.0.0.1:59454
[ZMQTerminalIPythonApp] Connecting to: tcp://127.0.0.1:59454
[ZMQTerminalIPythonApp] connecting stdin channel to tcp://127.0.0.1:42679
[ZMQTerminalIPythonApp] Connecting to: tcp://127.0.0.1:42679
[ZMQTerminalIPythonApp] connecting heartbeat channel to tcp://127.0.0.1:41244
Header: {"msg_id":"e2ca7fdd-dafb-45e5-a04e-3f500b4115aa","date":"2016-04-12T19:30:38.525379","session":"a4af1dae-266f-4244-a397-2f81041fb617","version":"5.0","username":"vh","msg_type":"kernel_info_request"}
Content: {}
Jupyter Console 4.1.1

[ZMQTerminalIPythonApp] Starting the jupyter console mainloop...
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'


ERROR: Kernel did not respond

Shutting down kernel
```

## notebook

```
vh@box ~ [i] % jupyter notebook --debug --ip=127.0.0.1
[D 08:47:04.159 NotebookApp] Searching ['/home/vh', '/home/vh/.jupyter', '/usr/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[D 08:47:04.159 NotebookApp] Looking for jupyter_config in /etc/jupyter
[D 08:47:04.160 NotebookApp] Looking for jupyter_config in /usr/local/etc/jupyter
[D 08:47:04.160 NotebookApp] Looking for jupyter_config in /usr/etc/jupyter
[D 08:47:04.160 NotebookApp] Looking for jupyter_config in /home/vh/.jupyter
[D 08:47:04.160 NotebookApp] Looking for jupyter_config in /home/vh
[D 08:47:04.161 NotebookApp] Looking for jupyter_notebook_config in /etc/jupyter
[D 08:47:04.161 NotebookApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
[D 08:47:04.161 NotebookApp] Looking for jupyter_notebook_config in /usr/etc/jupyter
[D 08:47:04.161 NotebookApp] Looking for jupyter_notebook_config in /home/vh/.jupyter
[D 08:47:04.161 NotebookApp] Looking for jupyter_notebook_config in /home/vh
[I 08:47:04.296 NotebookApp] Serving notebooks from local directory: /home/vh
[I 08:47:04.296 NotebookApp] 0 active kernels 
[I 08:47:04.296 NotebookApp] The Jupyter Notebook is running at: http://127.0.0.1:8888/
[I 08:47:04.296 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[D 08:47:04.506 NotebookApp] Using contents: services/contents
[D 08:47:04.536 NotebookApp] 200 GET /tree (127.0.0.1) 31.29ms
[D 08:47:04.706 NotebookApp] 304 GET /custom/custom.css (127.0.0.1) 5.00ms
[D 08:47:04.913 NotebookApp] 200 GET /static/services/contents.js?v=20160413084704 (127.0.0.1) 0.96ms
[D 08:47:04.923 NotebookApp] 200 GET /custom/custom.js?v=20160413084704 (127.0.0.1) 1.04ms
[D 08:47:04.970 NotebookApp] 200 GET /api/config/tree?_=1460530024847 (127.0.0.1) 0.96ms
[D 08:47:04.977 NotebookApp] 200 GET /api/config/common?_=1460530024848 (127.0.0.1) 0.98ms
[D 08:47:04.986 NotebookApp] 200 GET /api/terminals?_=1460530024849 (127.0.0.1) 0.66ms
[D 08:47:04.992 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:04.993 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:04.993 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:04.993 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:04.994 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:04.994 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:04.995 NotebookApp] 200 GET /api/kernelspecs (127.0.0.1) 4.20ms
[D 08:47:04.997 NotebookApp] 200 GET /api/sessions?_=1460530024850 (127.0.0.1) 1.33ms
[D 08:47:05.000 NotebookApp] 200 GET /api/terminals?_=1460530024851 (127.0.0.1) 0.63ms
[D 08:47:05.044 NotebookApp] 200 GET /api/contents?type=directory&_=1460530024852 (127.0.0.1) 11.90ms
[I 08:47:09.008 NotebookApp] Creating new notebook in 
[D 08:47:09.009 NotebookApp] Saving /home/vh/Untitled1.ipynb
[D 08:47:09.039 NotebookApp] 201 POST /api/contents (127.0.0.1) 32.28ms
[D 08:47:09.107 NotebookApp] Using contents: services/contents
[D 08:47:09.164 NotebookApp] 200 GET /notebooks/Untitled1.ipynb?kernel_name=haskell (127.0.0.1) 57.83ms
[D 08:47:09.265 NotebookApp] 304 GET /static/components/MathJax/MathJax.js?config=TeX-AMS_HTML-full,Safe&delayStartupUntil=configured (127.0.0.1) 0.69ms
[D 08:47:09.280 NotebookApp] 304 GET /custom/custom.css (127.0.0.1) 0.93ms
[D 08:47:09.415 NotebookApp] 304 GET /static/components/MathJax/config/TeX-AMS_HTML-full.js?rev=2.5.3 (127.0.0.1) 0.74ms
[D 08:47:09.606 NotebookApp] 304 GET /static/components/MathJax/config/Safe.js?rev=2.5.3 (127.0.0.1) 0.69ms
[D 08:47:09.687 NotebookApp] 304 GET /custom/custom.js?v=20160413084704 (127.0.0.1) 0.72ms
[D 08:47:09.721 NotebookApp] 200 GET /api/config/notebook?_=1460530029528 (127.0.0.1) 0.72ms
[D 08:47:09.726 NotebookApp] 200 GET /api/config/common?_=1460530029529 (127.0.0.1) 0.74ms
[D 08:47:09.805 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:09.805 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:09.806 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:09.806 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:09.806 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:09.806 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:09.807 NotebookApp] 304 GET /api/kernelspecs (127.0.0.1) 2.71ms
[D 08:47:09.845 NotebookApp] 200 GET /api/contents/Untitled1.ipynb?type=notebook&_=1460530029530 (127.0.0.1) 14.66ms
[D 08:47:09.847 NotebookApp] 200 GET /nbextensions/widgets/notebook/js/extension.js?v=20160413084704 (127.0.0.1) 0.80ms
[D 08:47:09.848 NotebookApp] 304 GET /static/components/MathJax/jax/output/HTML-CSS/fonts/STIX-Web/fontdata.js?rev=2.5.3 (127.0.0.1) 0.63ms
[D 08:47:09.879 NotebookApp] 304 GET /static/components/MathJax/extensions/Safe.js?rev=2.5.3 (127.0.0.1) 0.66ms
[D 08:47:10.013 NotebookApp] 200 GET /api/contents/Untitled1.ipynb/checkpoints?_=1460530029531 (127.0.0.1) 0.97ms
[D 08:47:10.023 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/init.js?v=20160413084704 (127.0.0.1) 0.88ms
[D 08:47:10.026 NotebookApp] 200 GET /nbextensions/widgets/notebook/js/widgetarea.js?v=20160413084704 (127.0.0.1) 0.90ms
[D 08:47:10.064 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:10.064 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:10.065 NotebookApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-0e7b53ee-5d5d-470d-b98c-e760ae715f4c.json', '--debug', '--ghclib', '/usr/lib/ghc']
[D 08:47:10.071 NotebookApp] Connecting to: tcp://127.0.0.1:56735
[I 08:47:10.072 NotebookApp] Kernel started: 0e7b53ee-5d5d-470d-b98c-e760ae715f4c
[D 08:47:10.072 NotebookApp] Kernel args: {'cwd': '/home/vh', 'kernel_name': 'haskell'}
[D 08:47:10.074 NotebookApp] 201 POST /api/sessions (127.0.0.1) 26.17ms
[D 08:47:10.091 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/manager.js?v=20160413084704 (127.0.0.1) 1.43ms
[D 08:47:10.094 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget.js?v=20160413084704 (127.0.0.1) 1.57ms
[D 08:47:10.099 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_link.js?v=20160413084704 (127.0.0.1) 2.02ms
[D 08:47:10.107 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_bool.js?v=20160413084704 (127.0.0.1) 7.00ms
[D 08:47:10.110 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_button.js?v=20160413084704 (127.0.0.1) 1.89ms
[D 08:47:10.112 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_box.js?v=20160413084704 (127.0.0.1) 1.45ms
[D 08:47:10.116 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_float.js?v=20160413084704 (127.0.0.1) 1.45ms
[D 08:47:10.119 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_image.js?v=20160413084704 (127.0.0.1) 1.54ms
[D 08:47:10.123 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_int.js?v=20160413084704 (127.0.0.1) 1.43ms
[D 08:47:10.126 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_color.js?v=20160413084704 (127.0.0.1) 1.38ms
[D 08:47:10.130 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_output.js?v=20160413084704 (127.0.0.1) 1.54ms
[D 08:47:10.135 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_selection.js?v=20160413084704 (127.0.0.1) 2.34ms
[D 08:47:10.140 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_selectioncontainer.js?v=20160413084704 (127.0.0.1) 2.31ms
[D 08:47:10.144 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_string.js?v=20160413084704 (127.0.0.1) 1.50ms
[D 08:47:10.148 NotebookApp] 200 GET /nbextensions/widgets/widgets/js/widget_controller.js?v=20160413084704 (127.0.0.1) 1.94ms
[D 08:47:10.162 NotebookApp] Found kernel haskell in /home/vh/.local/share/jupyter/kernels
[D 08:47:10.162 NotebookApp] Native kernel (python3) available from /usr/local/lib/python3.5/dist-packages/ipykernel/resources
[D 08:47:10.163 NotebookApp] Serving kernel resource from: /home/vh/.local/share/jupyter/kernels/haskell
[D 08:47:10.164 NotebookApp] 200 GET /kernelspecs/haskell/kernel.js?v=20160413084704 (127.0.0.1) 3.10ms
[D 08:47:10.281 NotebookApp] 200 GET /static/components/codemirror/addon/mode/loadmode.js?v=20160413084704 (127.0.0.1) 1.32ms
[D 08:47:10.330 NotebookApp] 200 GET /static/components/backbone/backbone-min.js?v=20160413084704 (127.0.0.1) 1.37ms
[D 08:47:10.347 NotebookApp] 200 GET /static/components/codemirror/mode/haskell/haskell.js?v=20160413084704 (127.0.0.1) 1.23ms
[D 08:47:10.353 NotebookApp] Using contents: services/contents
[W 08:47:10.364 NotebookApp] 404 GET /static/components/codemirror/mode/ihaskell/ihaskell.js?v=20160413084704 (127.0.0.1) 12.00ms referer=http://127.0.0.1:8888/notebooks/Untitled1.ipynb?kernel_name=haskell
[D 08:47:10.389 NotebookApp] Initializing websocket connection /api/kernels/0e7b53ee-5d5d-470d-b98c-e760ae715f4c/channels
[D 08:47:10.398 NotebookApp] Requesting kernel info from 0e7b53ee-5d5d-470d-b98c-e760ae715f4c
[D 08:47:10.399 NotebookApp] Connecting to: tcp://127.0.0.1:60900
Header: {"msg_id":"cb59192d-a54e-4bec-a788-77196e2cb78b","version":"5.0","date":"2016-04-13T08:47:10.401381","session":"EC87BF6960274D7C8F629AF16C655F67","username":"vh","msg_type":"kernel_info_request"}
Content: {}
[D 08:47:10.427 NotebookApp] 304 GET /nbextensions/widgets/widgets/css/widgets.min.css (127.0.0.1) 1.35ms
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'
[I 08:47:13.074 NotebookApp] KernelRestarter: restarting kernel (1/5)
[D 08:47:13.075 NotebookApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-0e7b53ee-5d5d-470d-b98c-e760ae715f4c.json', '--debug', '--ghclib', '/usr/lib/ghc']
[D 08:47:13.079 NotebookApp] Connecting to: tcp://127.0.0.1:56735
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'
[I 08:47:16.082 NotebookApp] KernelRestarter: restarting kernel (2/5)
[D 08:47:16.083 NotebookApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-0e7b53ee-5d5d-470d-b98c-e760ae715f4c.json', '--debug', '--ghclib', '/usr/lib/ghc']
[D 08:47:16.088 NotebookApp] Connecting to: tcp://127.0.0.1:56735
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'
[I 08:47:19.091 NotebookApp] KernelRestarter: restarting kernel (3/5)
[D 08:47:19.092 NotebookApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-0e7b53ee-5d5d-470d-b98c-e760ae715f4c.json', '--debug', '--ghclib', '/usr/lib/ghc']
[D 08:47:19.097 NotebookApp] Connecting to: tcp://127.0.0.1:56735
[W 08:47:20.402 NotebookApp] Timeout waiting for kernel_info reply from 0e7b53ee-5d5d-470d-b98c-e760ae715f4c
[D 08:47:20.403 NotebookApp] Opening websocket /api/kernels/0e7b53ee-5d5d-470d-b98c-e760ae715f4c/channels
[D 08:47:20.403 NotebookApp] Connecting to: tcp://127.0.0.1:60900
[D 08:47:20.403 NotebookApp] Connecting to: tcp://127.0.0.1:58030
[D 08:47:20.404 NotebookApp] Connecting to: tcp://127.0.0.1:49246
Header: {"version":"5.0","msg_id":"75B1197E9E3E43599C2C12864EB3E2E9","session":"EC87BF6960274D7C8F629AF16C655F67","username":"username","msg_type":"kernel_info_request"}
Content: {}
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'
[I 08:47:22.097 NotebookApp] KernelRestarter: restarting kernel (4/5)
WARNING:root:kernel 0e7b53ee-5d5d-470d-b98c-e760ae715f4c restarted
[D 08:47:22.099 NotebookApp] Starting kernel: ['/home/vh/.cabal/bin/ihaskell', 'kernel', '/run/user/1000/jupyter/kernel-0e7b53ee-5d5d-470d-b98c-e760ae715f4c.json', '--debug', '--ghclib', '/usr/lib/ghc']
[D 08:47:22.103 NotebookApp] Connecting to: tcp://127.0.0.1:56735
ihaskell: /home/vh/.cabal/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.8-AY0Q64QxcQ28d5fOASgof6/libHScryptonite-0.8-AY0Q64QxcQ28d5fOASgof6.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.8'
[W 08:47:25.103 NotebookApp] KernelRestarter: restart failed
[W 08:47:25.104 NotebookApp] Kernel 0e7b53ee-5d5d-470d-b98c-e760ae715f4c died, removing from map.
ERROR:root:kernel 0e7b53ee-5d5d-470d-b98c-e760ae715f4c restarted failed!
[W 08:47:25.180 NotebookApp] Kernel deleted before session
[W 08:47:25.180 NotebookApp] 410 DELETE /api/sessions/2a0520e6-3f55-40df-809c-d2839763a91d (127.0.0.1) 1.87ms referer=http://127.0.0.1:8888/notebooks/Untitled1.ipynb?kernel_name=haskell
```

# kernel.json

```
vh@box ~ [i] % cat $(jupyter --data-dir)/kernels/haskell/kernel.json
{"display_name":"Haskell","argv":["/home/vh/.cabal/bin/ihaskell","kernel","{connection_file}","--debug","--ghclib","/usr/lib/ghc"],"language":"haskell"}%   
```

If you need more information, please let me know.

**Comment by inktrap** [comment]
# Versions

## ghc and cabal
~~~
vh@box ~ [n] % ghc --numeric-version                                      
7.10.3
vh@box ~ [n] % cabal --version                                            
cabal-install version 1.22.6.0
using version 1.22.5.0 of the Cabal library 
~~~

## zeromq

~~~
vh@box ~ [i] % dpkg -l libzmq3-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                      Version                   Architecture              Description
+++-=========================================-=========================-=========================-========================================================================================
ii  libzmq3-dev:amd64                         4.1.4-7                   amd64                     lightweight messaging kernel (development files)
~~~

## Python

~~~
vh@box ~ [i] % pip3 freeze | egrep 'jupyter|ipython|zmq|tornado|Jinja2'
Jinja2==2.8
ipython==4.1.2
ipython-genutils==0.1.0
jupyter==1.0.0
jupyter-client==4.2.2
jupyter-console==4.1.1
jupyter-core==4.1.0
pyzmq==15.2.0
tornado==4.3
zmq==0.0.0
~~~

**Comment by inktrap** [comment]
@gibiansky any idea why this keeps happening?

**Comment by Hodapp87** [comment]
This looks maybe related to https://github.com/gibiansky/IHaskell/issues/636

**Comment by gibiansky** [comment]
Likely duplicate of #636. Closing in favor of that issue.

**Comment by ExpHP** [comment]
I am not certain; the *cause* of the error between these issues may be the same (or intimately related), but it looks like the *fixes* are independent. #636 is to be addressed by an update to stack, but **@inktrap's setup is 100% cabal.**

For reference, I too have run into this issue, and have worked around it by installing everything in a sandbox with a `./cabal.config` that contains

```
shared = True
executable-dynamic = True
```

as mentioned on the (purportedly resolved?) [ghc issue](https://ghc.haskell.org/trac/ghc/ticket/12147). Seeing as an update to stack fixes the other issue, it seems likely that the ball is in cabal's court rather than ihaskell's; but I do not personally understand the issue well enough to be sure.

**Comment by ExpHP** [comment]
:unamused: Installing `Cabal` 1.24 and `cabal-install` to the sandbox before everything else **did not make a difference**.  I still see these messages:

```
[W 14:15:40.274 NotebookApp] 404 GET /static/components/codemirror/mode/ihaskell/ihaskell.js?v=20160911141536 (127.0.0.1) 2.33ms referer=http://localhost:8888/notebooks/Bearz.ipynb
ihaskell: /home/lampam/ihaskell/sandbox/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.3/cryptonite-0.19-9fTgDVqlngxExANMsHUBS7/libHScryptonite-0.19-9fTgDVqlngxExANMsHUBS7.a: unhandled ELF relocation(RelA) type 42

ihaskell: ihaskell: unable to load package `cryptonite-0.19'
```

[I've posted notes on this recent misadventure in a gist](https://gist.github.com/ExpHP/cbec4a4d4e805be0501887c0aa3b330b)... but they probably aren't too helpful.


Notably, **I do not have this issue when using the Docker image.**
